isamples_metadata/taxonomy/SESARClassifierInput.py

```diff
@@ -79,20 +80,24 @@ def __init__(self, thing):
         self.SESAR_material_field = "material"
         self.SESAR_sample_field = "sampleType"
 
-    def build_text(self, description_map, labelType):
+    def build_text(self, description_map, labelType, field_order):
         """Return the concatenated text of informative fields in the
         description_map"""
         concatenated_text = ""
-        for key, value in description_map.items():
-            if key == "igsnPrefix":
-                continue
-            elif value == "":
-                # empty field
-                continue
-            elif labelType == "material" and key != self.SESAR_material_field:
-                concatenated_text += value + " , "
-            elif labelType == "sample" and key != self.SESAR_sample_field:
-                concatenated_text += value + " , "
+        # use the order of the fields to build text
+        # to have consistent text content
+        for field in field_order:
+            if field in description_map:
```

Comment:
Instead of extracting text fields from the dictionary `description_map,` fix the order of the fields that are going to be extracted using `field_order` when building the text.